packages/core/lib/command.js

```diff
@@ -101,48 +99,43 @@ class Command {
       }
     });
 
-    // Check unsupported command line flag according to the option list in help
-    try {
-      // while in `console` & `develop`, input is passed as a string, not as an array
-      if (!Array.isArray(inputStrings)) inputStrings = inputStrings.split(" ");
-      // Method `extractFlags(args)` : Extracts the `--option` flags from arguments
-      const inputOptions = extractFlags(inputStrings);
-      const validOptions = result.command.help.options
-        .map(item => {
-          let opt = item.option.split(" ")[0];
-          return opt.startsWith("--") ? opt : null;
-        })
-        .filter(item => {
-          return item != null;
-        });
-
-      let invalidOptions = inputOptions.filter(
-        opt => !validOptions.includes(opt)
-      );
+    // while in `console` & `develop`, input is passed as a string, not as an array
+    if (!Array.isArray(inputStrings)) inputStrings = inputStrings.split(" ");
+    // Method `extractFlags(args)` : Extracts the `--option` flags from arguments
+    const inputOptions = extractFlags(inputStrings);
+    const validOptions = result.command.help.options
+      .map(item => {
+        let opt = item.option.split(" ")[0];
+        return opt.startsWith("--") ? opt : null;
+      })
+      .filter(item => {
+        return item != null;
+      });
 
-      // TODO: Remove exception for 'truffle run' when plugin options support added.
-      if (invalidOptions.length > 0 && result.name !== "run") {
-        if (options.logger) {
-          const log = options.logger.log || options.logger.debug;
-          log(
-            "> Warning: possible unsupported (undocumented in help) command line option: " +
-              invalidOptions
-          );
-        }
+    let invalidOptions = inputOptions.filter(
+      opt => !validOptions.includes(opt)
+    );
+
+    // TODO: Remove exception for 'truffle run' when plugin options support added.
+    if (invalidOptions.length > 0 && result.name !== "run") {
+      if (options.logger) {
+        const log = options.logger.log || options.logger.debug;
+        log(
+          "> Warning: possible unsupported (undocumented in help) command line option: " +
+            invalidOptions
+        );
       }
+    }
 
-      const newOptions = Object.assign({}, clone, argv);
+    const newOptions = Object.assign({}, clone, argv);
 
-      result.command.run(newOptions, callback);
+    analytics.send({
+      command: result.name ? result.name : "other",
+      args: result.argv._,
+      version: bundled || "(unbundled) " + core
+    });
 
-      analytics.send({
-        command: result.name ? result.name : "other",
-        args: result.argv._,
-        version: bundled || "(unbundled) " + core
-      });
-    } catch (err) {
-      callback(err);
-    }
+    return result.command.run(newOptions);
```

Comment:
Actually it does matter apparently. @davidmurdoch pointed me to an interesting article about the subtle differences between these. I would love to throw in an `await`. If we don't throw one in here then I think we'll get a worse stacktrace or one that is not so useful since it won't throw in this context...I think that was the tl;dr.

Check this [cool thing](https://github.com/goldbergyoni/nodebestpractices/blob/master/sections/errorhandling/returningpromises.md) out!

Reply:
:grimacing: `await` removal was my fault in other areas of Truffle's code base because I was operating on the assumption that `await` behavior wouldn't change between Node versions. haha